drivers/sensor/lis2dh/lis2dh_trigger.c

```diff
@@ -407,56 +401,54 @@ int lis2dh_init_interrupt(const struct device *dev)
 #endif
 
 	/* data ready int1 gpio configuration */
-	status = gpio_pin_configure(lis2dh->gpio_int1, cfg->irq1_pin,
-				    GPIO_INPUT | cfg->irq1_flags);
+	status = gpio_pin_configure_dt(&cfg->gpio_drdy, GPIO_INPUT);
 	if (status < 0) {
-		LOG_ERR("Could not configure gpio %d", cfg->irq1_pin);
+		LOG_ERR("Could not configure gpio %d", cfg->gpio_drdy.pin);
 		return status;
 	}
 
 	gpio_init_callback(&lis2dh->gpio_int1_cb,
 			   lis2dh_gpio_int1_callback,
-			   BIT(cfg->irq1_pin));
+			   BIT(cfg->gpio_drdy.pin));
 
-	status = gpio_add_callback(lis2dh->gpio_int1, &lis2dh->gpio_int1_cb);
+	status = gpio_add_callback(cfg->gpio_drdy.port, &lis2dh->gpio_int1_cb);
 	if (status < 0) {
 		LOG_ERR("Could not add gpio int1 callback");
 		return status;
 	}
 
-	LOG_INF("int1 on %s.%02u", cfg->irq1_dev_name, cfg->irq1_pin);
+	LOG_INF("int1 on %s.%02u", cfg->gpio_drdy.port->name, cfg->gpio_drdy.pin);
 
 	/*
 	 * Setup INT2 (for Any Motion) if defined in DT
 	 */
+
 	/* setup any motion gpio interrupt */
-	lis2dh->gpio_int2 = device_get_binding(cfg->irq2_dev_name);
-	if (lis2dh->gpio_int2 == NULL) {
-		LOG_INF("Cannot get pointer to irq2_dev_name");
+	if (!device_is_ready(cfg->gpio_int.port)) {
```

Comment:
Is it correct for NULL ports?

Reply:
It is returning false, but not giving any error (status = 0)

Reply:
> It is returning false, but not giving any error (status = 0)

Then why LOG_ERR?

Reply:
> LOG_DBG

Sounds good

Reply:
ok, then. Thanks for your patience.

Reply:
Thanks for yours!